libr/fs/p/fs_r2.c

```diff
@@ -260,8 +260,9 @@ static RList *__cfg(RFSRoot *root, const char *path) {
 }
 
 static RList *__root(RFSRoot *root, const char *path) {
+	r_return_val_if_fail (root, NULL);
 	RList *list = r_list_newf (NULL);
-	r_return_val_if_fail (root && list, NULL);
+	r_return_val_if_fail (list, NULL);
```

Comment:
If list fails to be allocated it will be at tuntime so better use normal if check instead of compile time macro